core/async/src/test_loop.rs

```diff
@@ -371,33 +372,26 @@ impl<Data, Event: Debug + Send + 'static> TestLoop<Data, Event> {
     }
 
     /// Processes the given event, by logging a line first and then finding a handler to run it.
-    fn process_event(&mut self, mut event: EventInHeap<Event>) {
+    fn process_event(&mut self, event: EventInHeap) {
         let start_json = serde_json::to_string(&EventStartLogOutput {
             current_index: event.id,
             total_events: self.next_event_index,
-            current_event: format!("{:?}", event.event),
+            current_event: format!("{:?}", event.description),
             current_time_ms: event.due.whole_milliseconds() as u64,
         })
         .unwrap();
-        info!(target: "test_loop", "TEST_LOOP_EVENT_START {}", start_json);
+        tracing::info!(target: "test_loop", "TEST_LOOP_EVENT_START {}", start_json);
         assert_eq!(self.current_time, event.due);
 
-        for handler in &mut self.handlers {
-            if let Err(e) = handler.handle(event.event, &mut self.data) {
-                event.event = e;
-            } else {
-                // Push any new events into the queue. Do this before emitting the end log line,
-                // so that it contains the correct new total number of events.
-                self.queue_received_events();
-                let end_json = serde_json::to_string(&EventEndLogOutput {
-                    total_events: self.next_event_index,
-                })
+        (event.callback)(&mut self.data);
```

Comment:
We don't need to iterate through the set of handlers. Just call the callback function here instead.